### PR TITLE
[CLEANUP] - Remove `get_` from all getters

### LINKED ISSUE #3104: [CLEANUP] - Remove `get_` from all getters
### What is this task and why do we need to work on it?

Based on https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter, we should remove `get_` from all getters.

### What work will need to be done to complete this task?

_No response_

### Are there any other details to include?

_No response_

### What are the acceptance criteria to close this issue?

Getter names no longer start with `get`.

### Branch work will be merged to (if not the default branch)

_No response_

### PR DESCRIPTION
**Pull request by shenkeyao**
Closes #3104.

### This PR: 
* Removes `get_` from getters except for:
  * `get_or_calc_vid_share`, because it's either get or calculate.
  * `VidScheme` implementations, because it's defined in jellyfish.
* Removes duplicated functions that differ only in names.

### Key places to review: 
* Whether it makes sense to remove `get` from each name.

### CODE REVIEW

**Review by QuentinI — APPROVED** [review]
Looks great! Thanks for doing this

File: crates/hotshot/src/lib.rs [hunk 1]
```diff
@@ -146,7 +146,7 @@ pub struct SystemContext<TYPES: NodeType, I: NodeImplementation<TYPES>> {
 
     /// Access to the output event stream.
     #[deprecated(
-        note = "please use the `get_event_stream` method on `SystemContextHandle` instead. This field will be made private in a future release of HotShot"
+        note = "please use the `event_stream` method on `SystemContextHandle` instead. This field will be made private in a future release of HotShot"
     )]
     pub output_event_stream: (Sender<Event<TYPES>>, InactiveReceiver<Event<TYPES>>),
```
Comment (ss-es) [inline crates/hotshot/src/lib.rs:151]: I think this can probably be removed now

**Review by jparr721 — APPROVED** [review]

File: crates/orchestrator/src/client.rs [hunk 2]
```diff
@@ -193,7 +193,7 @@ impl OrchestratorClient {
     /// # Errors
     /// If we were unable to serialize the Libp2p data
     #[allow(clippy::type_complexity)]
-    pub async fn get_config_without_peer<K: SignatureKey>(
+    pub async fn config_without_peer<K: SignatureKey>(
```
Comment (ss-es) [inline crates/orchestrator/src/client.rs:196]: I think this should actually be `get_config_without_peer`. It's not a getter method, and I think `get` here is in the sense of `GET` as an HTTP request.

I would probably just revert the changes to the orchestrator, I think all the `get_` functions here are similar.

**Review by ss-es — CHANGES_REQUESTED** [review]
requesting changes for the orchestrator

File: crates/hotshot/src/types/handle.rs [hunk 3]
```diff
@@ -47,7 +47,7 @@ pub struct SystemContextHandle<TYPES: NodeType, I: NodeImplementation<TYPES>> {
 
 impl<TYPES: NodeType, I: NodeImplementation<TYPES> + 'static> SystemContextHandle<TYPES, I> {
     /// obtains a stream to expose to the user
-    pub fn get_event_stream(&self) -> impl Stream<Item = Event<TYPES>> {
+    pub fn event_stream(&self) -> impl Stream<Item = Event<TYPES>> {
```
[the remark this answers is not in the document]
Comment (ss-es) [reply crates/hotshot/src/types/handle.rs:50]: (so this is probably fine, was just thinking out loud)

File: crates/hotshot/src/lib.rs (hunk 1 above)
Comment (shenkeyao) [reply crates/hotshot/src/lib.rs:151]: Just to double-check my understanding. Did you mean that we can remove `pub` and delete this note, or that we can actually remove this field?

File: crates/orchestrator/src/client.rs (hunk 2 above)
Comment (shenkeyao) [reply crates/orchestrator/src/client.rs:196]: Good catch! Yeah, I think it makes more sense to keep the `get` for these functions.

File: crates/hotshot/src/lib.rs (hunk 1 above)
Comment (shenkeyao) [reply crates/hotshot/src/lib.rs]: Updated assuming it's the former case, since it looks like we can't just remove this function.

Comment (ss-es) [reply crates/hotshot/src/lib.rs]: Oh yes, sorry! I think I had in my head "just delete it", but you're right -- it's a necessary field, so we just need to make it private.
